pkg/patterns/declarative/reconciler.go

```diff
@@ -337,6 +362,43 @@ func (r *Reconciler) validateOptions() error {
 	return nil
 }
 
+// Expect reflect.Kind of obj is reflect.Ptr
+func (r *Reconciler) setReconcileMetrics(obj DeclarativeObject) (err error) {
+	var gvks []schema.GroupVersionKind
+	scheme := r.mgr.GetScheme()
+
+	if gvks, _, err = scheme.ObjectKinds(obj); err != nil {
+		return err
+	}
+
+	if len(gvks) != 1 {
+		for _, check := range gvks[1:] {
+			if gvks[0].Group != check.Group {
+				return fmt.Errorf("%s is registered to multiple groups in manager's scheme: %s, %s",
+					reflect.TypeOf(obj).Elem().Name(), gvks[0].Group, check.Group)
+			}
+		}
+
+		if vp := scheme.PrioritizedVersionsForGroup(gvks[0].Group); len(vp) != 0 {
+			for _, gv := range vp {
+				for _, check := range gvks {
+					if gv.Version == check.Version {
+						r.rc = reconcileCounterFor(check)
+						r.rfc = reconcileFailureCounterFor(check)
+						return
+					}
+				}
+			}
+		}
+		// Are there any function to recognize K8s API versioning?
```

Comment:
I'm not sure, @justinsb are you aware of anything "upstream" to resolve the api version here?

Reply:
Yes.

When
* `Scheme.ObjectKinds` returns multiple `GruopVersionKind` &
* `Scheme.PrioritizedVersionsForGroup` doesn't help determine which version to use,

we need a way to recognize K8s API versioning & choose appropriate (more matured) version.